src/arsenic/connection.py

```diff
@@ -88,7 +90,7 @@ def __init__(self, session: ClientSession, prefix: str):
         full_url = self.prefix + url
         log.info("request", url=strip_auth(full_url), method=method, body=body)
         async with self.session.request(
-            url=full_url, method=method, data=body
+            url=full_url, method=method, data=body, timeout=timeout or 300
```

Comment:
An odd question — what happens to the connection on timeout?
It is usable? Must it be closed?

Reply:
It throws an asyncio.TimeoutError for the request, but the session can still be used.